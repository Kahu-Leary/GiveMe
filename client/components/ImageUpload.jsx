import React, { useState } from 'react'
require("regenerator-runtime/runtime")


function ImageUpload() {
    const [image, setImage] = useState('')
    const [loading, setLoading] = useState(false)

    const uploadImage = async e => {
        const files = e.target.files
        const data = new FormData()
        data.append('file', files[0])
        data.append('upload_preset', 'GiveMe')
        setLoading(true)
        const res = await fetch(
            'https://api.cloudinary.com/v1_1/djogj7231/image/upload',
            {
                method: 'POST',
                body: data
            }
        )
        const file = await res.json()

        setImage(file.secure_url)
        setLoading(false)
    }

    return (
        <div className='ImageUpload'>
            <h5>Upload Image</h5>
            <input type='file'
                name='file'
                placeholder='Upload an Image'
                onChange={uploadImage}
                />
                {loading ? (
                    <h6>Loading...</h6>
                ) : (
                    <img src={image} style={{width: '300px'}} />
                )}
        </div>
    )
}

export default ImageUpload