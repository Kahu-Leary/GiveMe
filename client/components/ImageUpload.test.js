import React from 'react'

import renderer from 'react-test-renderer'

import ImageUpload from './ImageUpload'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ImageUpload/>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
