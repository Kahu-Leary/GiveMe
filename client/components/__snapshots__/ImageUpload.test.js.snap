// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ImageUpload"
>
  <input
    name="file"
    onChange={[Function]}
    placeholder="Upload an Image"
    type="file"
  />
  <img
    src=""
    style={
      Object {
        "width": "300px",
      }
    }
  />
</div>
`;
